UCR.Plugins/Remapper/AxisToAxisDamped.cs

```diff
@@ -0,0 +1,144 @@
+﻿using System;
+using System.Timers;
+using HidWizards.UCR.Core.Attributes;
+using HidWizards.UCR.Core.Models;
+using HidWizards.UCR.Core.Models.Binding;
+using HidWizards.UCR.Core.Utilities;
+using HidWizards.UCR.Core.Utilities.AxisHelpers;
+
+namespace HidWizards.UCR.Plugins.Remapper
+{
+    [Plugin("Axis to Axis (Damped)")]
+    [PluginInput(DeviceBindingCategory.Range, "Axis")]
+    [PluginOutput(DeviceBindingCategory.Range, "Axis")]
+    public class AxisToAxisDamped : Plugin
+    {
+        [PluginGui("Invert", ColumnOrder = 0)]
+        public bool Invert { get; set; }
+
+        [PluginGui("Linear", ColumnOrder = 3)]
+        public bool Linear { get; set; }
+
+        [PluginGui("Dead zone", ColumnOrder = 1)]
+        public int DeadZone { get; set; }
+
+        [PluginGui("Sensitivity", ColumnOrder = 2)]
+        public int Sensitivity { get; set; }
+
+        /// <summary>
+        /// Framerate of the smoothing simulation.
+        /// High framerates make the simulation faster and more responsive,
+        /// but also takes up more resources.
+        /// </summary>
+        [PluginGui("Refresh rate", ColumnOrder = 0, RowOrder = 2)]
+        public int FPS { get; set; }
+
+        /// <summary>
+        /// Weight of an imaginary object that's attached to a spring.
+        /// Heavy objects move slowly and take a while to react to sudden 
+        /// changes in input. Light objects react immediately to any changes.
+        /// Value should not be 0, otherwise there's a division by zero problem.
+        /// </summary>
+        [PluginGui("Weight", ColumnOrder = 1, RowOrder = 2)]
+        public int Weight { get; set; }
+
+        /// <summary>
+        /// Damping factor. High values reduce the spring's movement a lot.
+        /// Small values let the spring move for a long time. Value should
+        /// be between 0% and a 100%. Other values make the simulation unstable.
+        /// </summary>
+        [PluginGui("Damping", ColumnOrder = 2, RowOrder = 2)]
+        public int Damping { get; set; }
+
+        private float _targetValue;
+        private float _velocity;
```

Comment:
I'm afraid I can't do that. The _velocity variable is global and carries velocity information from one simulation frame to the next.

https://github.com/Snoothy/UCR/blob/ab0a044b23f59b45facca22b9372946c0fff2489/UCR.Plugins/Remapper/AxisToAxisDamped.cs#L89-L90

Reply:
_When the value of a private field is always assigned to in a class' methods before being read, then it is not being used to store class information._

If I understand correctly, this:

```
_velocity += acceleration; 
```

is wrong, but this:

```
_velocity = _velocity + acceleration; 
```

would be correct because it's being read first? Can I maybe leave this one as it is, because I think it looks a lot nicer? Or is it required to get rid of these things...?